service/history/workflow/mutable_state_impl.go

```diff
@@ -4409,6 +4410,9 @@ func (ms *MutableStateImpl) AddWorkflowExecutionOptionsUpdatedEvent(
 
 func (ms *MutableStateImpl) ApplyWorkflowExecutionOptionsUpdatedEvent(event *historypb.HistoryEvent) error {
 	override := event.GetWorkflowExecutionOptionsUpdatedEventAttributes().GetVersioningOverride()
+	if event.GetWorkflowExecutionOptionsUpdatedEventAttributes().GetUnsetVersioningOverride() {
+		override = nil
+	}
```

Comment:
You need to check if `override == nil && !unsetVersioningOverride`. In this case, you don't want to do anything, right?

```suggestion
	attributes := event.GetWorkflowExecutionOptionsUpdatedEventAttributes()
	override := attributes.GetVersioningOverride()
	if attributes.GetUnsetVersioningOverride() {
		override = nil
	} else if override == nil {
		return
	}
```

Reply:
omg thank you so much for catching that!